contrib/bindings/go/pathrs/capi.go

```diff
@@ -0,0 +1,178 @@
+// Copyright (C) 2019, 2020 Aleksa Sarai <cyphar@cyphar.com>
+// Copyright (C) 2019, 2020 SUSE LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package pathrs
+
+/*
+#cgo LDFLAGS:-L ../../../../target/debug/ -l pathrs
+#include "../../../../include/pathrs.h"
+*/
+import "C"
+import (
+	"unsafe"
+)
+
+type pointerType = int
+
+const (
+	pointerOnNone   pointerType = C.PATHRS_NONE
+	pointerOnError  pointerType = C.PATHRS_ERROR
+	pointerOnRoot   pointerType = C.PATHRS_ROOT
+	pointerOnHandle pointerType = C.PATHRS_HANDLE
+)
+
+type resolverType = uint64
+
+const (
+	kernelResolver   resolverType = C.PATHRS_KERNEL_RESOLVER
+	emulatedResolver resolverType = C.PATHRS_EMULATED_RESOLVER
+)
+
+type cRootPtr = unsafe.Pointer
+
+type cErrorPtr = unsafe.Pointer
+
+type cHandlePtr = unsafe.Pointer
+
+type fd = int
+
+type errorType struct {
+	savedErrno  C.uint64_t
+	description *C.char
+	backtrace   *errorBacktrace
+}
+
+type errorBacktrace struct {
+	head   *backtraceEntry
+	length C.uintptr_t
+}
+
+type backtraceEntry struct {
+	ip            uintptr
+	symbolAddress uintptr
+	symbolName    *C.char
+	symbolFile    *C.char
+	symbolLineno  C.uint32_t
+	// TODO: This field I found it to be generated by `go tool cgo`
+	// But after it be deleted nothing is changed but it has to be
+	// It should be investigate more dipply
+	// _             [4]byte
```

Comment:
This is padding added to the structure. Technically it is necessary to include, but since it's at the end it doesn't make too much difference. I'm probably going to redesign how structure passing works with libpathrs because I **really** don't like `pathrs_configure` right now -- we might switch to a string-based configuration system.